Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+# busybox:v1.36.0
+FROM busybox@sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5 as release
+
+ARG PRODUCT_NAME=nomad
```

Comment:
I think this FROM might not be correct. When I look at, it looks like the `amd64` specific image and not multi-arch manifest for the tag. But in your build.yml changes I think your intent was to create arm64 & amd64 docker images.

For that, I think you need to use the manifest digest.

```
# only one arch here, doesn't appear to me a manifest list
▸ docker buildx imagetools inspect docker.io/library/busybox@sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5
Name:      docker.io/library/busybox@sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5
MediaType: application/vnd.docker.distribution.manifest.v2+json
Digest:    sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5

# recommended change
▸ docker buildx imagetools inspect docker.io/library/busybox:1.36.0
Name:      docker.io/library/busybox:1.36.0
MediaType: application/vnd.docker.distribution.manifest.list.v2+json
Digest:    sha256:9e2bbca079387d7965c3a9cee6d0c53f4f4e63ff7637877a83c4c05f2a666112

Manifests:
  Name:      docker.io/library/busybox:1.36.0@sha256:086417a48026173aaadca4ce43a1e4b385e8e62cc738ba79fc6637049674cac0
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/amd64

  Name:      docker.io/library/busybox:1.36.0@sha256:57e3ef7ce54066c75a2e0b05210717452f5471093b27a5b2114e43dbf541261a
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v5

  Name:      docker.io/library/busybox:1.36.0@sha256:946858083a14c5bcdb0287d8b29e59fd064567c0454e1f6b7e5f988f7b84e6a1
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v6

  Name:      docker.io/library/busybox:1.36.0@sha256:efe5726a54c25a584068095ab1f68cde55ed5a43d19fbefa5e4de9ef7d8f8546
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v7

  Name:      docker.io/library/busybox:1.36.0@sha256:1af0a016aaa24dd9ddf2ff0771160a68d6394eff88c9c0d5239c93a21ea06f9f
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm64/v8

  Name:      docker.io/library/busybox:1.36.0@sha256:096958b089cdfa4b345dba0ae0a1e43bea59e4de6e084c26429b6c85096322cf
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/386

  Name:      docker.io/library/busybox:1.36.0@sha256:eff4a3502698259caf724b1e0ee44c5269eff595af99a188e2f288b5efdfc372
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/mips64le

  Name:      docker.io/library/busybox:1.36.0@sha256:440d25bf9cfa8ba300233ca4e5199a2e35f1ec952af5edc4244c51972f955a52
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/ppc64le

  Name:      docker.io/library/busybox:1.36.0@sha256:b817c03cceeb7bd721cb984fcea521d680fa3a34b3027a203d6995696e462fc4
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/riscv64

  Name:      docker.io/library/busybox:1.36.0@sha256:bbb13843fa881b390b058c5f27f1b199781c2aa1443619e7a1be818b04c2ff7e
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/s390x
```

But when I test it (just to double-check) I do see differences:
Given:
```
$ cat Dockerfile
FROM busybox@sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5 
#FROM docker.io/library/busybox@sha256:9e2bbca079387d7965c3a9cee6d0c53f4f4e63ff7637877a83c4c05f2a666112
RUN uname -a
```
I see the same image arch regardless of the platform,:
```
▸ docker build --platform linux/amd64 --progress=plan --no-cache -f Dockerfile  . 2>&1 | grep Linux
#5 0.167 Linux buildkitsandbox 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 x86_64 GNU/Linux

▸ docker build --platform linux/arm64 --progress=plan --no-cache -f Dockerfile  . 2>&1 | grep Linux
#6 0.191 Linux buildkitsandbox 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 x86_64 GNU/Linux
```

But if I use the manifest list (swap my FROM comment):
```
▸ docker build --platform linux/amd64 --progress=plan --no-cache -f Dockerfile  . 2>&1 | grep Linux
#5 0.175 Linux buildkitsandbox 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 x86_64 GNU/Linux

▸ docker build --platform linux/arm64 --progress=plan --no-cache -f Dockerfile  . 2>&1 | grep Linux
#0 0.133 Linux buildkitsandbox 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 aarch64 GNU/Linux
```

```suggestion
# docker.io/library/busybox:1.36.0
# When pinning use the multi-arch manifest list, `docker buildx imagetools inspect ...`
FROM docker.io/library/busybox@sha256:9e2bbca079387d7965c3a9cee6d0c53f4f4e63ff7637877a83c4c05f2a666112
as release

ARG PRODUCT_NAME=nomad
```

PS I admit, I'm confused as to where the `acaddd` digest is coming from. I cannot find it any any reference to 1.36.0 images.

Reply:
Ah thank you, that SHA came from here: https://github.com/hashicorp/nomad/pull/17017#discussion_r1187960672

I will update it and see how it goes 😄